openmc/mgxs/groups.py

```diff
@@ -0,0 +1,275 @@
+
+        Parameters
+        ----------
+        group : Integral
+            The energy group index, starting at 1 for the highest energies
+
+        Returns
+        -------
+        2-tuple
+            The low and high energy bounds for the group in MeV
+
+        Raises
+        ------
+        ValueError
+            If the group edges have not yet been set.
+
+        """
+
+        if self.group_edges is None:
+            msg = 'Unable to get energy group bounds for group "{0}" since ' \
+                  'the group edges have not yet been set'.format(group)
+            raise ValueError(msg)
+
+        lower = self.group_edges[self.num_groups-group]
+        upper = self.group_edges[self.num_groups-group+1]
+        return lower, upper
+
+    def get_group_indices(self, groups='all'):
+        """Returns the array indices for one or more energy groups.
+
+        Parameters
+        ----------
+        groups : str, tuple
+            The energy groups of interest - a tuple of the energy group indices,
+            starting at 1 for the highest energies (default is 'all')
+
+        Returns
+        -------
+        ndarray
+            The ndarray array indices for each energy group of interest
+
+        Raises
+        ------
+        ValueError
+            If the group edges have not yet been set, or if a group is requested
+            that is outside the bounds of the number of energy groups.
+
+        """
+
+        if self.group_edges is None:
+            msg = 'Unable to get energy group indices for groups "{0}" since ' \
+                  'the group edges have not yet been set'.format(groups)
+            raise ValueError(msg)
+
+        if groups == 'all':
+            indices = np.arange(self.num_groups)
+        else:
+            indices = np.zeros(len(groups), dtype=np.int)
+
+        for i, group in enumerate(groups):
+            cv.check_greater_than('group', group, 0)
+            cv.check_less_than('group', group, self.num_groups, equality=True)
+            indices[i] = group - 1
+
+        return indices
+
+    def get_condensed_groups(self, coarse_groups):
+        """Return a coarsened version of this EnergyGroups object.
+
+        This method merges together energy groups in this object into wider
+        energy groups as defined by the list of groups specified by the user,
+        and returns a new, coarse EnergyGroups object.
+
+        Parameters
+        ----------
+        coarse_groups : Iterable of 2-tuple
+            The energy groups of interest - a list of 2-tuples, each directly
+            corresponding to one of the new coarse groups. The values in the
+            2-tuples are upper/lower energy groups used to construct a new
+            coarse group. For example, if [(1,2), (2,4)] was used as the coarse
+            groups, fine groups 1 and 2 would be merged into coarse group 1
+            while fine groups 3 and 4 would be merged into coarse group 2.
```

Comment:
I have a feeling something is wrong with the example here. If `(1,2)` means combine groups 1-2, how does `(2,4)` mean combined groups 3-4?